src/data/finset/basic.lean

```diff
@@ -526,6 +526,15 @@ protected theorem induction {α : Type*} {p : finset α → Prop} [decidable_eq
     { rw [insert_val, ndinsert_of_not_mem m] }
   end) nd
 
+protected theorem induction' {α : Type*} {p : finset α → Prop}
+  (h₁ : p ∅) (h₂ : ∀ ⦃a : α⦄ {s : finset α} (h : a ∉ s), p s → p (cons a s h)) : ∀ s, p s
+| ⟨s, nd⟩ := multiset.induction_on s (λ _, h₁) (λ a s IH nd, begin
+    cases nodup_cons.1 nd with m nd',
+    rw [← (eq_of_veq _ : cons a (finset.mk s _) m = ⟨a ::ₘ s, nd⟩)],
+    { exact h₂ (by exact m) (IH nd') },
+    { rw [cons_val] }
+  end) nd
+
```

Comment:
Can you use this definition to simplify the `induction` lemma above? It can now be proved as:
```
protected theorem induction {α : Type*} {p : finset α → Prop} [decidable_eq α]
  (h₁ : p ∅) (h₂ : ∀ ⦃a : α⦄ {s : finset α}, a ∉ s → p s → p (insert a s)) : ∀ s, p s :=
finset.induction' h₁ $ λ a s ha, (s.cons_eq_insert a ha).symm ▸ h₂ ha
```

Reply:
Yes I can certainly do that if you prefer it. I don't know what `@[elab_as_eliminator]` means though, I guess I'll just leave the one that's there where it is?

Reply:
You should add `@[elab_as_eliminator]` to your lemma too